src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKReadFilterPluginDescriptor.java

```diff
@@ -33,38 +35,28 @@ private void readObject(java.io.ObjectInputStream in)
     private static final String pluginPackageName = "org.broadinstitute.hellbender.engine.filters";
     private static final Class<?> pluginBaseClass = org.broadinstitute.hellbender.engine.filters.ReadFilter.class;
 
-    @Argument(fullName = StandardArgumentDefinitions.READ_FILTER_LONG_NAME,
-            shortName = StandardArgumentDefinitions.READ_FILTER_SHORT_NAME,
-            doc="Read filters to be applied before analysis", optional=true)
-    public final List<String> userReadFilterNames = new ArrayList<>(); // preserve order
-
-    @Argument(fullName = StandardArgumentDefinitions.DISABLE_READ_FILTER_LONG_NAME,
-            shortName = StandardArgumentDefinitions.DISABLE_READ_FILTER_SHORT_NAME,
-            doc="Read filters to be disabled before analysis", optional=true)
-    public final Set<String> disableFilters = new HashSet<>();
-
-    @Argument(fullName = "disableAllReadFilters",
-            shortName = "disableAllReadFilters",
-            doc = "Disable all read filters", common = false, optional = true)
-    public boolean disableAllReadFilters = false;
+    @ArgumentCollection
+    protected final GATKReadFilterPluginArgumentCollection args;
 
     // Map of read filter (simple) class names to the corresponding discovered plugin instance
-    private Map<String, ReadFilter> readFilters = new HashMap<>();
+    protected Map<String, ReadFilter> readFilters = new HashMap<>();
```

Comment:
I changed the access level in the second commit of some parts of the code to make the descriptor extensible. There are some design issues for my personal tools and usage that I would like to solve in my project without affecting GATK, like:
* I would like to have a more verbose help in the plugin, and the returned sets of `getAllowedValuesForDescriptorArgument` should output in my case 1) the values of default tool filters for disable then and 2) all the possible filter names (found by reflection) for the user to know the possible values.
* The treatment of `disableAllReadFilters` to disable just the defaults but not the provided by the user.
* I need the returned filters of `getAllInstances` to return also the tool defaults.

This was the main point of my work in this plugin, because I don't want to copy everything and modify a couple of methods, but just extend the class to provide my own method. In addition, this will be for me also a temporary solution for the issue regarding the plugin package, where I should provide my own one.